### ISSUE TITLE
state initializer not working correctly for some negative values

### ISSUE THREAD

**Issue by alfrisch** [Qiskit/qiskit-terra#1721]
### Informations

- **Qiskit Terra version**: master
- **Python version**: 3.7.1
- **Operating system**: Win10

### What is the current behavior?

When trying to initialize a state using `initialize` for a vector with some specific negative entries, e.g. `[-0.5, 0.5, 0.5, 0.5]`, the state has wrong signs, e.g. it turns out to be `[0.5, -0.5, -0.5, -0.5]`.
This can not be checked using an assertion on fidelity as it is a global phase.
In case of `[0.5, -0.5, 0.5, 0.5]` this error does not appear.

### Steps to reproduce the problem

```
from qiskit import QuantumCircuit
from qiskit import QuantumRegister
from qiskit import execute, BasicAer
from qiskit.quantum_info import state_fidelity

desired_vector = [-0.5, 0.5, 0.5, 0.5]
qr = QuantumRegister(2, "qr")
qc = QuantumCircuit(qr)
qc.initialize(desired_vector, [qr[0], qr[1]])
job = execute(qc, BasicAer.get_backend('statevector_simulator'))
result = job.result()
statevector = result.get_statevector()
fidelity = state_fidelity(statevector, desired_vector)
```

### What is the expected behavior?

The `statevector` should show the same signs as `desired_vector`.

### Suggested solutions

The bug might be caused in `InitializeGate` method `_rotations_to_disentangle`, but I did not investigate further.

**Comment by ismaila-at-za-ibm** [comment]
Thanks @alfrisch, for this. I am working on it and have made progress. The problems is not in `_rotations_to_disentangle` but in the fact that the global phase was intentionally not added.

**Comment by jaygambetta** [comment]
This is not an error or error. Global phase does not matter.

**Comment by jaygambetta** [comment]
@ajavadia we should close this and add to documentation or something.

**Comment by ismaila-at-za-ibm** [comment]
I agree that global phase has no measurable effect on a quantum state. However it may be useful to match the global phase of the requested initialization vector for testing purposes.
PR:
Addressed Issue #1721: Introduced a Gate for Global Phase adjustment
Designed a new gate that results in global phase change. Even though global phase has no measurable difference to a quantum outcome, it may be useful for simulation purposes.

**Comment by jaygambetta** [comment]
see comments in the pr.
